v03_pipeline/lib/reference_data/compare_globals.py

```diff
@@ -86,30 +87,21 @@ def from_ht(
 
 
 def get_datasets_to_update(
-    rdc: ReferenceDatasetCollection,
     ht1_globals: Globals,
     ht2_globals: Globals,
-    dataset_type: DatasetType,
 ) -> list[str]:
-    return [
-        dataset
-        for dataset in rdc.datasets(dataset_type)
-        if not validate_globals_match(rdc, ht1_globals, ht2_globals, dataset)
-    ]
-
+    datasets_to_update = set()
 
-def validate_globals_match(
-    rdc: ReferenceDatasetCollection,
-    ht1_globals: Globals,
-    ht2_globals: Globals,
-    dataset: str,
-) -> bool:
-    results = []
     for field in dataclasses.fields(Globals):
-        result = ht1_globals[field.name].get(dataset) == ht2_globals[field.name].get(
-            dataset,
-        )
-        if result is False:
-            logger.info(f'{field.name} mismatch for {dataset}, {rdc.value}')
-        results.append(result)
-    return all(results)
+        ht1_datasets = set(ht1_globals[field.name].keys())
+        ht2_datasets = set(ht2_globals[field.name].keys())
+        datasets_to_update.update(ht1_datasets.symmetric_difference(ht2_datasets))
```

Comment:
nit, but maybe the set ops are cleaner?  `^` for symmetric difference and `&` for intersection?